OpenRA.Mods.Common/Traits/World/BridgeLayer.cs

```diff
@@ -57,5 +80,69 @@ public void Remove(Actor b)
 				if (bridges[c] == b)
 					bridges[c] = null;
 		}
+
+		public void WorldLoaded(World w, WorldRenderer wr)
+		{
+			if (info.Bridges.Length > 0)
+			{
+				// Build a list of templates that should be overlaid with bridges.
+				foreach (var bridge in info.Bridges)
+				{
+					var bi = w.Map.Rules.Actors[bridge].TraitInfo<BridgeInfo>();
+					foreach (var template in bi.Templates)
+						bridgeTypes.Add(template.Template, (bridge, template.Health));
+				}
+
+				// Take all templates to overlay from the map.
+				foreach (var cell in w.Map.AllCells.Where(cell => bridgeTypes.ContainsKey(w.Map.Tiles[cell].Type)))
+					ConvertBridgeToActor(w, cell);
+			}
+		}
+
+		void ConvertBridgeToActor(World w, CPos cell)
+		{
+			// This cell already has a bridge overlaying it from a previous iteration.
+			if (bridges[cell] != null)
+				return;
+
+			// Correlate the tile "image" aka subtile with its position to find the template origin.
+			var ti = w.Map.Tiles[cell];
+			var tile = ti.Type;
+			var index = ti.Index;
+			var template = terrainInfo.Templates[tile];
+			var ni = cell.X - index % template.Size.X;
+			var nj = cell.Y - index / template.Size.X;
+
+			// Create a new actor for this bridge and keep track of which subtiles this bridge includes.
+			var bridge = w.CreateActor(bridgeTypes[tile].Template, new TypeDictionary
+			{
+				new LocationInit(new CPos(ni, nj)),
```

Comment:
What is `ni` and `nj`?

Reply:
Copy pasted, I'm not sure myself. It looks like a location from which we offset

Reply:
renamed